pkgs/development/mobile/androidenv/deploy-androidpackages.nix

```diff
@@ -0,0 +1,59 @@
+{stdenv, lib, unzip, mkLicenses}:
+{packages, os ? null, buildInputs ? [], patchesInstructions ? {}, meta ? {}, ...}@args:
+
+let
+  extraParams = removeAttrs args [ "packages" "os" "buildInputs" "patchesInstructions" ];
+  sortedPackages = builtins.sort (x: y: builtins.lessThan x.name y.name) packages;
+in
+stdenv.mkDerivation ({
+  pname = lib.concatMapStringsSep "-" (package: package.name) sortedPackages;
+  version = lib.concatMapStringsSep "-" (package: package.revision) sortedPackages;
+  src = map (package:
+    if os != null && builtins.hasAttr os package.archives then package.archives.${os} else package.archives.all
+  ) packages;
+  buildInputs = [ unzip ] ++ buildInputs;
+  preferLocalBuild = true;
+
+  unpackPhase = ''
+    buildDir=$PWD
+    i=0
+    for srcArchive in $src; do
+      extractedZip="extractedzip-$i"
+      i=$(($i+1))
```

Comment:
i=$((i+1))

Reply:
I copied it from Internet :D I had to force push because I changed the commit message, so I fixed them in that new commit. Here it's https://github.com/NixOS/nixpkgs/commit/bceccb7f707c0b0cfba7288cfcb11c4832ac1272